lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -54,26 +56,73 @@ internal sealed class FileSystemInputPartition<T> : IInputPartition<T>, IDisposa
             _id = id;
             _fileSystem = fileSystem;
             _fileSerializer = fileSerializer;
-            _filePaths = filePaths;
             _tempFileCreator = tempFileCreator;
-            _isInitialized = false;
-            _copyToLocal = copyToLocal;
+            _filePaths = filePaths;
+
+            if (!copyToLocal)
+            {
+                // Implies that the files are already local.
```

Comment:
In original code, files are provided in _filePaths, usually remote files. 

If the user what to download files to local first, and if the files are not downloaded yet, then download (Initialize()), then Deserialize from the local downloaded files. 

Else, meaning if user doesn't choose to copyToLocal, then Deserialize from the original files directly.